tests/components/electric_kiwi/test_sensor.py

```diff
@@ -0,0 +1,92 @@
+"""The tests for Electric Kiwi sensors."""
+
+
+from datetime import UTC, datetime
+from unittest.mock import AsyncMock, Mock, patch
+
+from freezegun import freeze_time
+import pytest
+
+from homeassistant.components.electric_kiwi import (
+    DOMAIN,
+    ElectricKiwiHOPDataCoordinator,
+)
+from homeassistant.components.electric_kiwi.const import ATTRIBUTION
+from homeassistant.components.electric_kiwi.sensor import _check_and_move_time
+from homeassistant.components.sensor import SensorDeviceClass
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import ATTR_ATTRIBUTION, ATTR_DEVICE_CLASS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_registry import EntityRegistry
+import homeassistant.util.dt as dt_util
+
+from .conftest import TIMEZONE
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.parametrize(
+    ("sensor", "sensor_state"),
+    [
+        ("sensor.hour_of_free_power_start", "4:00 PM"),
+        ("sensor.hour_of_free_power_end", "5:00 PM"),
+    ],
+)
+async def test_hop_sensors(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    ek_api: AsyncMock,
+    entity_registry: EntityRegistry,
+    sensor: str,
+    sensor_state: str,
+) -> None:
+    """Test HOP sensors for the Electric Kiwi integration.
+
+    This time (note no day is given, its only a time) is fed from the Electric Kiwi API.
+    if the API returns 4:00 PM, the sensor state should be set to today at 4pm or if now is past 4pm, then tomorrow at 4pm
+
+    """
+
+    with patch(
+        "homeassistant.helpers.config_entry_oauth2_flow.async_get_config_entry_implementation",
+        return_value=AsyncMock(),
+    ):
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+        entries = hass.config_entries.async_entries(DOMAIN)
+        assert entries[0].state is ConfigEntryState.LOADED
```

Comment:
I'd use the config_entry variable here.

```suggestion
        assert config_entry.state is ConfigEntryState.LOADED
```